jquery.highlight.js

```diff
@@ -69,62 +105,146 @@
     });
   };
 
+  // Calculates the current viewport
+  //
+  // @return [Oobject] current viewport properties
+  var getViewPort = function () {
+    var top = window.scrollY ||
+              window.pageYOffset ||
+              document.documentElement.scrollTop;
+    var left = window.scrollX ||
+               window.pageXOffset ||
+               document.documentElement.scrollLeft;
+    var bottom = top + (
+                   window.innerHeight ||
+                   document.documentElement.clientHeight
+                 );
+    var right = left + (
+                  window.innerWidth ||
+                  document.documentElement.clientWidth
+                );
+    return {
+      top: top,
+      left: left,
+      bottom: bottom,
+      right: right,
+      height: bottom - top,
+      width: right - left
+    }
+  }
+
+  // Calculates the given element offset
+  //
+  // @return [Object] current viewport properties
+  var getElementOffset = function (element) {
+    var viewport = getViewPort()
+    var offset = $(element).offset();
+    var top = offset.top - viewport.top;
+    var left = offset.left - viewport.left;
+    var bottom = top + $(element).outerHeight();
+    var right = left + $(element).outerWidth();
+    if (top < 0) { top = 0; }
+    if (left < 0) { top = 0; }
+    if (right < 0) { top = 0; }
+    if (bottom < 0) { top = 0; }
+    return {
+      top: top,
+      left: left,
+      bottom: bottom,
+      right: right,
+      height: bottom - top,
+      width: right - left
+    }
+  }
+
+  var isElementInViewport = function (element) {
```

Comment:
Can you comment what the difference between `isElementInPage` vs `isElementInViewport`?